packages/datetime/src/dateInput.tsx

```diff
@@ -205,8 +205,20 @@ export class DateInput extends AbstractPureComponent2<IDateInputProps, IDateInpu
         const { value, valueString } = this.state;
         const dateString = this.state.isInputFocused ? valueString : getFormattedDateString(value, this.props);
         const dateValue = isDateValid(value) ? value : null;
-        const dayPickerProps = {
+        const dayPickerProps: DayPickerProps = {
             ...this.props.dayPickerProps,
+            // If the user tabs on a DayPicker Day element and the lastElementInPopover is also a DayPicker Day
```

Comment:
I think this could be a little more clear, making it explicit that it's TAB being pressed while the day is focussed, not the TAB moving focus _onto_ the day element

```suggestion
            // If the user presses the TAB key on a DayPicker Day element and the lastElementInPopover is also a DayPicker Day
```